### PR TITLE
Cover more trick assignments

### PR DESCRIPTION
**Pull request by ddatsko**
### What does this PR do?
I added some more tests to Nuitka/tests/basics/TrickAssignments32 where was a comment with TODO. Also made "__int\__" method return anobject of type ```int``` instead of ```bool``` as I was getting warning while running tests.
### Why was it initiated? 
I just wanted to make a contribution to take one of Nuitka's projects as my GSoC project and while examining code I saw this TODO comment and found out that this job is suitable for me.(I'm sorry if that was the job of someone else, but the file haven`t been changed for 26 days, so I decided to try)

### PR Checklist
- [x] Correct base branch selected? `develop` for new features and bug fixes too. If it's
      part of a hotfix, it will be moved to ``master`` during it.
- [x] All tests still pass. Check the developer manual about ``Running the Tests``. There
      are Travis tests that cover the most important things however, and you are
      welcome to rely on those, but they might not cover enough.
- [x] Ideally new features or fixed regressions ought to be covered via new tests.
- [x] Ideally new or changed features have documentation updates.

### CODE REVIEW

File: tests/basics/TrickAssignments32.py [hunk 1]
```diff
@@ -510,7 +510,7 @@ def __int__(self):
             nonlocal a
             del a
 
-            return False
```
Comment (kayhayen) [inline tests/basics/TrickAssignments32.py]: What warning were you talking about? I tended to purposefully no return booleans, as that must work as well.

File: tests/basics/TrickAssignments32.py [hunk 2]
```diff
@@ -1047,4 +1047,172 @@ def __bytes__(self):
     print("Ouch.!")
 
 
+def someFunctionThatReturnsDeletedValueViaEq():
```
Comment (kayhayen) [inline tests/basics/TrickAssignments32.py]: Can you do the same to `__lt__`, `__lte__`, `__gt__`, `__gte__`, `__neq__` (spelling ?) too?

Yours,
Kay

File: tests/basics/TrickAssignments32.py (hunk 1 above)
Comment (ddatsko) [reply tests/basics/TrickAssignments32.py]:
```
DeprecationWarning: __int__ returned non-int (type bool).  The ability to return an instance of a strict subclass of int is deprecated, and may be removed in a future version of Python.
  c = C()
```
This is the message I get when using both Nuitka and default Python interpreter.
```
Python 3.7.3
```

File: tests/basics/TrickAssignments32.py (hunk 2 above)
Comment (ddatsko) [reply tests/basics/TrickAssignments32.py]: Sure, I will do this

File: tests/basics/TrickAssignments32.py (hunk 1 above)
Comment (ddatsko) [reply tests/basics/TrickAssignments32.py]: @kayhayen , so, should I change this back or its OK?

Comment (kayhayen) [reply tests/basics/TrickAssignments32.py]: I meant to say something about this, the test runner ignores warnings. The test code often does bad things, but Nuitka is not about compiling good code, but one that actually exists. You ought to use `./run_all.py filename` to execute the test and get a comparative diff. To create the test, you should use standard Python first, make that work, then compare.

Comment (kayhayen) [reply tests/basics/TrickAssignments32.py]: And yes, please revert it, as it actually lowers test coverage.

Comment (ddatsko) [reply tests/basics/TrickAssignments32.py]: OK, thank you for explanation

**Pull request merged**